playwright/test/search-images.test.ts

```diff
@@ -28,41 +25,22 @@ test('(1) | Search by term, filter by colour, check results, view image details,
   await newSearch(context, page, 'images');
   await searchQuerySubmitAndWait('art of science', page);
 
-  if (isMobile(page)) {
-    await page.locator('button[aria-controls="mobile-filters-modal"]').click();
-    await expect(await page.locator('#mobile-filters-modal')).toBeVisible();
-    await page.getByTestId('swatch-red').click();
-    await page.getByRole('button', { name: 'Close modal window' }).click();
-  } else {
-    await page.locator('button[aria-controls="images.color"]').click();
-    await page.getByRole('button', { name: 'Red', exact: true }).click();
-    await page.locator('button[aria-controls="images.color"]').click();
-  }
-  await expect(
-    await page.getByTestId('image-search-results-container')
-  ).toBeVisible();
+  await selectAndWaitForColourFilter(page);
   await expect(
-    await page
-      .locator('[data-testid="image-search-results-container"] li')
-      .first()
+    page.getByTestId('image-search-results-container')
   ).toBeVisible();
   await clickActionClickSearchResultItem(1, page);
-  await expect(
-    await page.locator('a[aria-label="View expanded image"]')
-  ).toBeVisible();
+  await expect(page.getByLabel('View expanded image')).toBeVisible();
 
   // Check we show visually similar images.  This could theoretically fail
   // if the first result doesn't have any similar images, but if it fails
   // it's much more likely we've broken something on the page.
-  await expectItemIsVisible('h3 >> text="Visually similar images"', page);
+  await expect(
+    page.getByRole('heading', { name: 'Visually similar images' })
+  ).toBeVisible();
 
-  await Promise.all([
-    safeWaitForNavigation(page),
-    page.click(
-      `div[id="expanded-image-dialog"] a[aria-label="View expanded image"]`
-    ),
-  ]);
-  expectUrlToMatch(regexImageGalleryUrl, page);
+  await page.getByLabel('View expanded image').click();
+  await expect(page).toHaveURL(RegExp(regexImageGalleryUrl));
```

Comment:
nitpick; Could we add a comment here just for future reference as to what it checks, since regexes are not always the most readable thing? 🙏

Reply:
or change the name of the variable to make it more transparent? maybe regexImageViewerUrl?

Reply:
It goes to the item viewer right? so maybe `ItemViewerURLRegex`?

I _think_ in this case looking at the URL makes sense, but I know @jamieparkinson[ had some thoughts about looking at URL](https://wellcome.slack.com/archives/C3TQSF63C/p1704205220271429?thread_ts=1704204194.038399&cid=C3TQSF63C). It's very much a navigational change in this case though so I think it's in line with what he meant by it.